pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go

```diff
@@ -450,22 +444,23 @@ func (cra *cpuResourceAdvisor) gc() {
 	}
 }
 
-func (cra *cpuResourceAdvisor) getContainerRegions(ci *types.ContainerInfo) ([]region.QoSRegion, bool) {
-	if v, ok := cra.containerRegionMap[ci.PodUID]; ok {
-		if regions, exist := v[ci.ContainerName]; exist {
-			return regions, true
+func (cra *cpuResourceAdvisor) getContainerRegions(ci *types.ContainerInfo) ([]region.QoSRegion, error) {
+	var regions []region.QoSRegion = nil
+	for regionName := range ci.RegionNames {
+		region, ok := cra.regionMap[regionName]
+		if !ok {
+			return nil, fmt.Errorf("failed to find region %v", regionName)
 		}
+		regions = append(regions, region)
 	}
-	return nil, false
+	return regions, nil
 }
 
 func (cra *cpuResourceAdvisor) setContainerRegions(ci *types.ContainerInfo, regions []region.QoSRegion) {
```

Comment:
This function will not change container info in metacache because it is called by metacache.RangeContainer, which is a reader function. Use RangeAndUpdateContainer instead.